src/main/java/nathanbot/commands/CommandHandler.java

```diff
@@ -0,0 +1,208 @@
+    private static final String EXIT = "Bye. Hope to see you again soon!\n";
+    private static final String GREET = """
+                        Hello! I'm NathanBot
+                        What can I do for you?
+                       """;
+
+    /**
+     * Handles the greet command by printing a greeting message.
+     */
+    public static void handleGreet() {
+        System.out.println(LINE + GREET + LINE);
+    }
+
+    /**
+     * Handles the exit command by printing a farewell message.
+     */
+    public static void handleExit() {
+        System.out.println(LINE + EXIT + LINE);
+    }
+
+    /**
+     * Handles the display list command by printing the task list.
+     *
+     * @param taskList The task list to display.
+     */
+    public static void handleDisplayList(TaskList taskList) {
+        System.out.println(LINE + taskList + LINE);
+    }
+
+    /**
+     * Handles the mark command, marking a task as done or undone.
+     *
+     * @param input The user input containing the command and task number.
+     * @param command The command string (e.g., "mark" or "unmark").
+     * @param taskList The list of tasks.
+     * @param isDone True if the task should be marked as done, false otherwise.
+     */
+    public static void handleMarkCommand(String input, String command, TaskList taskList, boolean isDone) {
+        // Logic implemented by me; syntax and formatting recommended by Copilot.
+        try {
+            int index = Integer.parseInt(input.substring(command.length()));
+            if (isDone) {
+                taskList.markAsDone(index - 1);
+                System.out.println(LINE + "Nice! I've marked this task as done:\n  " + taskList.getTask(index - 1) + "\n" + LINE);
+            } else {
+                taskList.markAsUndone(index - 1);
+                System.out.println(LINE + "OK, I've marked this task as not done yet:\n  " + taskList.getTask(index - 1) + "\n" + LINE);
+            }
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            System.out.println(LINE + "Invalid task number. To see the list of tasks, use: list\n" + LINE);
+        }
+    }
+
+    /**
+     * Handles the delete command by removing a task from the task list.
+     *
+     * @param input The user input containing the command and task number.
+     * @param taskList The list of tasks.
+     */
+    public static void handleDeleteCommand(String input, TaskList taskList) {
+        // Logic implemented by me; syntax and formatting recommended by Copilot.
+        try {
+            int index = Integer.parseInt(input.substring(CommandType.DELETE.getCommand().length()));
+            Task task = taskList.getTask(index - 1);
+            taskList.deleteTask(index - 1);
+            System.out.println(LINE + "Noted. I've removed this task:\n  " + task + "\nNow you have " + taskList.listLength() + " tasks in the list.\n" + LINE);
```

Comment:
Wouldn't it be better to make this line into two so that it would not hit the max character of 120?